src/documentation.ts

```diff
@@ -68,8 +68,19 @@ export class Documentation extends OpenApiBuilder {
       : undefined;
   }
 
-  protected ensureUniqOperationId(path: string, method: Method) {
-    const operationId = makeCleanId(path, method);
+  protected ensureUniqOperationId(
+    path: string,
+    method: Method,
+    endpointOperationId?: string,
+  ) {
+    if (
+      endpointOperationId &&
+      endpointOperationId in this.lastOperationIdSuffixes
+    ) {
+      throw new Error(`Duplicated operationId: "${endpointOperationId}"`);
+    }
+
+    const operationId = endpointOperationId ?? makeCleanId(path, method);
```

Comment:
I'd suggest a slightly different logic here and above:

- if defined by user
  - if duplicated
    - throw
  - return what's defined by user
- continue the regular flow as it is unchanged (generating by `makeCleanId`)

Reply:
this is for keeping the user's choice as it is without any suffixes and avoid duplicated logic.
should be more clear